Can't compile DEploid cloned from https://github.com/DEploid-dev/DEploid.git

» git clone https://github.com/DEploid-dev/DEploid.git                              
Cloning into 'DEploid'...
remote: Enumerating objects: 4598, done.
remote: Total 4598 (delta 0), reused 0 (delta 0), pack-reused 4598
Receiving objects: 100% (4598/4598), 155.77 MiB | 31.43 MiB/s, done.
Resolving deltas: 100% (3333/3333), done.
 
» ls                                                                                
DEploid  DEploid-0.6-beta.1  v0.6-beta.1.tar.gz  v0.6-beta.2.tar.gz
 
» cd DEploid                                                                    
 
» ls                                                              
AUTHORS    CODE_OF_CONDUCT.md  COPYING       Dockerfile  Makefile.am               README     tests
bootstrap  configure.ac        data          docs        NEWS                      README.md  utilities
Brewfile   CONTRIBUTING.md     DEploid.wiki  m4          PULL_REQUEST_TEMPLATE.md  src        VERSION
 
» cd src                                                          
 
» ls                                                         
chooseK.cpp  dEploid.cpp     export      ibd.hpp   panel.cpp      txtReader.hpp  utility.hpp       vcfReader.hpp
chooseK.hpp  dEploidIO.cpp   global.hpp  lasso     panel.hpp      updateHap.cpp  variantIndex.cpp
codeCogs     dEploidIO.hpp   gzstream    mcmc.cpp  random         updateHap.hpp  variantIndex.hpp
debug        exceptions.hpp  ibd.cpp     mcmc.hpp  txtReader.cpp  utility.cpp    vcfReader.cpp
 
» ls lasso                                                
 
» cd ..                                                       
 
» pwd                                                             
/path/DEploid/DEploid
 
» ls                                                            
AUTHORS    CODE_OF_CONDUCT.md  COPYING       Dockerfile  Makefile.am               README     tests
bootstrap  configure.ac        data          docs        NEWS                      README.md  utilities
Brewfile   CONTRIBUTING.md     DEploid.wiki  m4          PULL_REQUEST_TEMPLATE.md  src        VERSION
 
» ./bootstrap                                                     
rm: cannot remove ‘Makefile’: No such file or directory
rm: cannot remove ‘Makefile.in’: No such file or directory
rm: cannot remove ‘autom4te.cache’: No such file or directory
rm: cannot remove ‘config.*’: No such file or directory
rm: cannot remove ‘depcomp’: No such file or directory
rm: cannot remove ‘install-sh’: No such file or directory
rm: cannot remove ‘missing’: No such file or directory
rm: cannot remove ‘configure’: No such file or directory
rm: cannot remove ‘aclocal.m4’: No such file or directory
rm: cannot remove ‘.deps’: No such file or directory
rm: cannot remove ‘INSTALL’: No such file or directory
rm: cannot remove ‘compile’: No such file or directory
rm: cannot remove ‘test-driver’: No such file or directory
rm: cannot remove ‘docs/doxygen/Makefile’: No such file or directory
rm: cannot remove ‘docs/doxygen/Makefile.in’: No such file or directory
configure.ac:18: installing './config.guess'
configure.ac:18: installing './config.sub'
configure.ac:4: installing './install-sh'
configure.ac:4: installing './missing'
Makefile.am:1: warning: shell git show HEAD | head -1 | sed -e "s/commit //g" | cat: non-POSIX variable name
Makefile.am:1: (probably a GNU make extension)
Makefile.am:2: warning: shell git submodule status | grep DEploid-Lasso-lib | sed -e "s/ //g" -e "s/DEploid.*//" | cat: non-POSIX variable name
Makefile.am:2: (probably a GNU make extension)
Makefile.am:8: warning: shell date -u | sed -e "s/ /-/g": non-POSIX variable name
Makefile.am:8: (probably a GNU make extension)
Makefile.am:90: warning: deprecated feature: target 'utilities' overrides 'utilities$(EXEEXT)'
Makefile.am:90: change your target to read 'utilities$(EXEEXT)'
/usr/share/automake-1.13/am/program.am: target 'utilities$(EXEEXT)' was defined here
Makefile.am:11:   while processing program 'utilities'
Makefile.am:90: warning: user target 'utilities$(EXEEXT)' defined here ...
/usr/share/automake-1.13/am/program.am: ... overrides Automake target 'utilities' defined here
Makefile.am:11:   while processing program 'utilities'
Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
Makefile.am:9: warning: user variable 'distdir' defined here ...
/usr/share/automake-1.13/am/distdir.am: ... overrides Automake variable 'distdir' defined here
checking for a BSD-compatible install... /bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '1387' is supported by ustar format... yes
checking whether GID '1079' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for ranlib... ranlib
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking for TestCase in -lcppunit... no
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for size_t... yes
./configure: line 5246: AX_CXXFLAGS_WARN_ALL: command not found
checking for doxygen... doxygen
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating docs/doxygen/Doxyfile
config.status: creating Makefile
config.status: creating docs/doxygen/Makefile
config.status: executing depfiles commands
 
» make                                                            
  CXX      src/dEploid-dEploid.o
  CXX      src/random/dEploid-fastfunc.o
  CXX      src/random/dEploid-random_generator.o
  CXX      src/random/dEploid-mersenne_twister.o
  CXX      src/dEploid-chooseK.o
  CXX      src/dEploid-ibd.o
  CXX      src/dEploid-mcmc.o
  CXX      src/dEploid-panel.o
  CXX      src/dEploid-utility.o
  CXX      src/dEploid-dEploidIO.o
src/dEploidIO.cpp:33:38: fatal error: lasso/src/dEploidLasso.hpp: No such file or directory
compilation terminated.
make: *** [src/dEploid-dEploidIO.o] Error 1
 
» ls src/lasso                                                    
src
 
» ls src/lasso/src                                                
 
» find /path/DEploid/ -name "dEploidLasso.hpp" -print
 
It is possible that I am using an outdated version of autoconf:
 
» module list                                                     
Currently Loaded Modulefiles:
  1) EasyBuild/2.9.0                                            9) OpenBLAS/0.2.18-GCC-5.4.0-2.26-LAPACK-3.6.1
  2) uge/8.5.5                                                 10) gompi/2016b
  3) GCCcore/5.4.0                                             11) FFTW/3.3.4-gompi-2016b
  4) binutils/2.26-GCCcore-5.4.0                               12) ScaLAPACK/2.0.2-gompi-2016b-OpenBLAS-0.2.18-LAPACK-3.6.1
  5) GCC/5.4.0-2.26                                            13) foss/2016b
  6) numactl/2.0.11-GCC-5.4.0-2.26                             14) doxygen/1.8.11-foss-2016b
  7) hwloc/1.11.3-GCC-5.4.0-2.26                               15) M4/1.4.17-GCC-5.4.0-2.26
  8) OpenMPI/1.10.3-GCC-5.4.0-2.26                             16) autoconf/2.69-GCC-5.4.0-2.26
 
Please let us know what you think.
 
Thanks!

And... there are more errors, if I bootstrap and make the missing DEploid-Lasso-lib library from https://github.com/shajoezhu/DEploid-Lasso-lib/tree/b229c805f9077660ec6a841352ccfca5ed0d8341 and link the src directory to either DEploid-Lasso-lib/src in DEploid-0.6-beta.1/DEploid-0.6-beta.2 or github's DEploid/src/lasso/src:
```
» make                                                           ~/eb_files/DEploid/DEploid-0.6-beta.2
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/dEploid-dEploid.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/random/dEploid-fastfunc.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/random/dEploid-random_generator.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/random/dEploid-mersenne_twister.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/dEploid-ibd.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/dEploid-mcmc.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/dEploid-panel.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/dEploid-utility.o
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /nethome)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
  CXX      src/dEploid-dEploidIO.o
In file included from src/lasso/dEploidLasso.hpp:31:0,
                 from src/dEploidIO.cpp:33:
src/lasso/lasso.hpp:66:12: error: redefinition of ‘template<class T> std::vector<_RealType> vecDiff(const std::vector<_RealType>&, const std::vector<_RealType>&)’
 vector <T> vecDiff(const vector<T> &vecA,
            ^
In file included from src/dEploidIO.cpp:29:0:
src/utility.hpp:40:12: note: ‘template<class T> std::vector<_RealType> vecDiff(const std::vector<_RealType>&, const std::vector<_RealType>&)’ previously declared here
 vector <T> vecDiff(const vector<T> &vecA, const vector<T> &vecB ) {
            ^
In file included from src/lasso/dEploidLasso.hpp:31:0,
                 from src/dEploidIO.cpp:33:
src/lasso/lasso.hpp:78:12: error: redefinition of ‘template<class T> std::vector<_RealType> vecSum(const std::vector<_RealType>&, const std::vector<_RealType>&)’
 vector <T> vecSum(const vector<T> &vecA, const vector<T> &vecB ) {
            ^
In file included from src/dEploidIO.cpp:29:0:
src/utility.hpp:58:12: note: ‘template<class T> std::vector<_RealType> vecSum(const std::vector<_RealType>&, const std::vector<_RealType>&)’ previously declared here
 vector <T> vecSum(const vector<T> &vecA, const vector<T> &vecB) {
            ^
In file included from src/lasso/dEploidLasso.hpp:31:0,
                 from src/dEploidIO.cpp:33:
src/lasso/lasso.hpp:89:12: error: redefinition of ‘template<class T> std::vector<_RealType> vecProd(const std::vector<_RealType>&, const std::vector<_RealType>&)’
 vector <T> vecProd(const vector<T> &vecA, const vector<T> &vecB) {
            ^
In file included from src/dEploidIO.cpp:29:0:
src/utility.hpp:69:12: note: ‘template<class T> std::vector<_RealType> vecProd(const std::vector<_RealType>&, const std::vector<_RealType>&)’ previously declared here
 vector <T> vecProd(const vector<T> &vecA,
            ^
make: *** [src/dEploid-dEploidIO.o] Error 1
```

This thread has been super helpful, as I've had similar problems as others while trying to compile DEploid. Manually compiling and replacing lasso in the src/ directory worked for me [as described by noamteyssier](https://github.com/DEploid-dev/DEploid/issues/309#issuecomment-521017780), however now that I'm past that, I'm getting a new error:

```bash
make: *** No rule to make target `utilities/dataExplore.r', needed by `utilities'.  Stop.
```

Can you advise me on what to do from here? I'm using macOS Mojave version 10.14.6.

--- a/.travis.yml
+++ b/.travis.yml
@@ -31,21 +31,16 @@ before_script:
   - ./bootstrap
 
 script:
-  - make
-  - make check
-  - if [ $TRAVIS_OS_NAME == linux ]; then ./tests/test_binary.sh; fi
-  - ./tests/testPOS.sh
-  - ./tests/test_binaryVcfVsTxt.sh
-  - ./tests/test-against-previous-version.sh
-  - ./tests/test_binaryReproducible.sh
-  - if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile; cd ../..; fi
   - rm -r *
   - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/issue309.tar.gz -o /dev/null
   - tar -xf issue309.tar.gz
-  - mv DEploid-issue309/* .
+  - cd DEploid-issue309
   - ./bootstrap
   - make; make check
-  - cd; cd DEploid-issue309
+  - cd ..; rm -r *
+  - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/issue309.tar.gz -o /dev/null
+  - tar -xf issue309.tar.gz
+  - mv DEploid-issue309/* .
   - ./bootstrap
   - make; make check
 after_success:

--- a/.travis.yml
+++ b/.travis.yml
@@ -40,8 +40,8 @@ script:
   - ./tests/test_binaryReproducible.sh
   - if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile; cd ../..; fi
   - rm -r *
-  - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/master.tar.gz -o /dev/null
-  - tar -xf master.tar.gz
+  - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/issue309.tar.gz -o /dev/null
+  - tar -xf issue309.tar.gz
   - mv DEploid-master/* .
   - ./bootstrap
   - make DEploid

--- a/.travis.yml
+++ b/.travis.yml
@@ -39,6 +39,11 @@ script:
   - ./tests/test-against-previous-version.sh
   - ./tests/test_binaryReproducible.sh
   - if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile; cd ../..; fi
-
+  - rm -r *
+  - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/dev.tar.gz -o /dev/null
+  - tar -xf dev.tar.gz
+  - mv DEploid-dev/* .
+  - ./bootstrap
+  - make DEploid
 after_success:
   - coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'

--- a/.travis.yml
+++ b/.travis.yml
@@ -31,6 +31,14 @@ before_script:
   - ./bootstrap
 
 script:
+  - make
+  - make check
+  - if [ $TRAVIS_OS_NAME == linux ]; then ./tests/test_binary.sh; fi
+  - ./tests/testPOS.sh
+  - ./tests/test_binaryVcfVsTxt.sh
+  - ./tests/test-against-previous-version.sh
+  - ./tests/test_binaryReproducible.sh
+  - if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile; cd ../..; fi
   - rm -r *
   - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/issue309.tar.gz -o /dev/null
   - tar -xf issue309.tar.gz

--- a/.travis.yml
+++ b/.travis.yml
@@ -42,8 +42,9 @@ script:
   - rm -r *
   - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/master.tar.gz -o /dev/null
   - tar -xf master.tar.gz
-  - mv DEploid-dev/* .
+  - mv DEploid-master/* .
   - ./bootstrap
   - make DEploid
+  - make check
 after_success:
   - coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'

--- a/.travis.yml
+++ b/.travis.yml
@@ -42,9 +42,9 @@ script:
   - rm -r *
   - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/issue309.tar.gz -o /dev/null
   - tar -xf issue309.tar.gz
-  - mv DEploid-master/* .
+  - mv DEploid-issue309/* .
   - ./bootstrap
-  - make DEploid
+  - make 
   - make check
 after_success:
   - coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'

--- a/.travis.yml
+++ b/.travis.yml
@@ -44,7 +44,9 @@ script:
   - tar -xf issue309.tar.gz
   - mv DEploid-issue309/* .
   - ./bootstrap
-  - make 
-  - make check
+  - make; make check
+  - cd; cd DEploid-issue309
+  - ./bootstrap
+  - make; make check
 after_success:
   - coveralls --exclude lib --exclude tests --exclude src/random --exclude src/codeCogs/ --exclude src/export/ --exclude src/gzstream/ --gcov-options '\-lp'

--- a/.travis.yml
+++ b/.travis.yml
@@ -41,7 +41,7 @@ script:
   - if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile; cd ../..; fi
   - rm -r *
   - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/master.tar.gz -o /dev/null
-  - tar -xf dev.tar.gz
+  - tar -xf master.tar.gz
   - mv DEploid-dev/* .
   - ./bootstrap
   - make DEploid

--- a/.travis.yml
+++ b/.travis.yml
@@ -40,7 +40,7 @@ script:
   - ./tests/test_binaryReproducible.sh
   - if [ $TRAVIS_OS_NAME == linux ]; then cd docs/doxygen; doxygen Doxyfile; cd ../..; fi
   - rm -r *
-  - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/dev.tar.gz -o /dev/null
+  - wget --no-check-certificate https://github.com/DEploid-dev/DEploid/archive/master.tar.gz -o /dev/null
   - tar -xf dev.tar.gz
   - mv DEploid-dev/* .
   - ./bootstrap

--- a/bootstrap
+++ b/bootstrap
@@ -2,9 +2,26 @@
 
 rm -r Makefile Makefile.in autom4te.cache config.* depcomp install-sh missing configure aclocal.m4 .deps INSTALL compile test-driver docs/doxygen/Makefile docs/doxygen/Makefile.in
 
+root_dir=$PWD
+if [ ! -d "${root_dir}/src" ]; then
+  "Error: please execute this script from DEploid root directory"
+  exit 1
+fi
+
+echo -n "Preparing ..."
+echo ""
+echo $root_dir/.git
+# First check if this is downloaded from zip or a git clone
+if [ -d ${root_dir}/.git ]; then
+	# use git submodule to down load files
+	git submodule update --init --recursive
+else
+	./checkfiles.sh
+fi # [-f ${root_dir}/.git]
+echo " done."
+
 
 aclocal
 autoconf
 automake -a
 ./configure
-

--- a/checkfiles.sh
+++ b/checkfiles.sh
@@ -13,7 +13,7 @@ fi
 OS=`uname -s`
 #echo "operating system is ${OS}"
 # VERSION=$(cat VERSION | tr -d '\n')
-VERSION=v0.1-release
+VERSION=0.1-release
 echo -n "." # echo "ok not git clone, now check plot and figure"
 
 #rm master.tar.gz  # just making sure download the one we actually want to
@@ -21,32 +21,31 @@ echo -n "." # echo "ok not git clone, now check plot and figure"
 extract_submodule_from(){
  account=$1
  submodule=$2
- plactto=$3
+ placeto=$3
  echo "Extract submodule $submodule from account $account, place at $3"
  if [ ! -d ${submodule}_dir ]; then mkdir ${submodule}_dir; fi
  echo -n "." # echo "ok, do something" #
 
  if [[ "${OS}" == "Linux" ]]; then
-   wget --no-check-certificate https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz -o /dev/null
+   wget --no-check-certificate https://github.com/${account}/${submodule}/archive/v${VERSION}.tar.gz -o /dev/null
  elif [[ "${OS}" == "Darwin"* ]]; then
-   curl -LOk https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz
+   curl -LOk https://github.com/${account}/${submodule}/archive/v${VERSION}.tar.gz
  else
-   echo "Unknown OS, fail to download package from https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz"
+   echo "Unknown OS, fail to download package from https://github.com/${account}/${submodule}/archive/v${VERSION}.tar.gz"
    echo "Please contact Joe at sha.joe.zhu@gmail.com if assistance is needed"
    exit 1
  fi
 ls *gz
 
- if [ ! -f ${VERSION}.tar.gz ]; then
-   echo "Error: Download package from https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz failed"
+ if [ ! -f v${VERSION}.tar.gz ]; then
+   echo "Error: Download package from https://github.com/${account}/${submodule}/archive/v${VERSION}.tar.gz failed"
    echo "Please contact Joe at sha.joe.zhu@gmail.com if assistance is needed"
    exit 1
  fi
- tar -xf ${VERSION}.tar.gz -C ${submodule}_dir
+ tar -xf v${VERSION}.tar.gz -C ${submodule}_dir
  if [ ! -d ${placeto} ]; then mkdir ${placeto}; fi
- # ls  ${submodule}_dir/*
- cp -r ${submodule}_dir/* ${placeto}/
- rm -r ${submodule}_dir ${VERSION}.tar.gz
+ cp -r ${submodule}_dir/${submodule}-${VERSION}/* ${placeto}/
+ rm -r ${submodule}_dir v${VERSION}.tar.gz
 }
 
 

--- a/checkfiles.sh
+++ b/checkfiles.sh
@@ -1,0 +1,71 @@
+#!/bin/bash
+
+#if [ ! -z "$src_dir+x" ]; then
+src_dir=$PWD/src
+#fi
+
+if [ ! -f "${src_dir}/dEploid.cpp" ]; then
+ "Error: please execute \".checkfiles.sh\" from DEploid root directory"
+ exit 1
+fi
+
+
+OS=`uname -s`
+#echo "operating system is ${OS}"
+# VERSION=$(cat VERSION | tr -d '\n')
+VERSION=v0.1-release
+echo -n "." # echo "ok not git clone, now check plot and figure"
+
+#rm master.tar.gz  # just making sure download the one we actually want to
+
+extract_submodule_from(){
+ account=$1
+ submodule=$2
+ plactto=$3
+ echo "Extract submodule $submodule from account $account, place at $3"
+ if [ ! -d ${submodule}_dir ]; then mkdir ${submodule}_dir; fi
+ echo -n "." # echo "ok, do something" #
+
+ if [[ "${OS}" == "Linux" ]]; then
+   wget --no-check-certificate https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz -o /dev/null
+ elif [[ "${OS}" == "Darwin"* ]]; then
+   curl -LOk https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz
+ else
+   echo "Unknown OS, fail to download package from https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz"
+   echo "Please contact Joe at sha.joe.zhu@gmail.com if assistance is needed"
+   exit 1
+ fi
+ls *gz
+
+ if [ ! -f ${VERSION}.tar.gz ]; then
+   echo "Error: Download package from https://github.com/${account}/${submodule}/archive/${VERSION}.tar.gz failed"
+   echo "Please contact Joe at sha.joe.zhu@gmail.com if assistance is needed"
+   exit 1
+ fi
+ tar -xf ${VERSION}.tar.gz -C ${submodule}_dir
+ if [ ! -d ${placeto} ]; then mkdir ${placeto}; fi
+ # ls  ${submodule}_dir/*
+ cp -r ${submodule}_dir/* ${placeto}/
+ rm -r ${submodule}_dir ${VERSION}.tar.gz
+}
+
+
+ACCOUNT=DEploid-dev
+SUBMODLUE=DEploid-Utilities
+PLACETO=utilities
+tmp_dir=$PWD/utilities
+if [ -f ${tmp_dir}/DEploidR.R  -a -f ${tmp_dir}/dEploidTools.r ];then
+ echo -n "." # echo "ok, do nothing"
+else
+ extract_submodule_from $ACCOUNT $SUBMODLUE $PLACETO
+fi
+
+ACCOUNT=shajoezhu
+SUBMODLUE=DEploid-Lasso-lib
+PLACETO=$PWD/src/lasso
+tmp_dir=$PWD/src/lasso
+if [ -f ${tmp_dir}/main.cpp  -a -f ${tmp_dir}/README.md ];then
+ echo -n "." # echo "ok, do nothing"
+else
+ extract_submodule_from $ACCOUNT $SUBMODLUE $PLACETO
+fi

